docs/Guides/zopen.md

```diff
@@ -0,0 +1,119 @@
+- `ZOPEN_DEPS`: a space-separated list of all software dependencies this package has. These packages will automatically be downloaded if they are not present in your $HOME/zopen/prod or $HOME/zopen/boot directories.
+
+To help guage the build quality of the port, a `zopen_check_results()` function needs to be provided inside the buildenv. This function should process
+the test results and emit a report of the failures, total number of tests, and expected number of failures to stdout as in the following format: 
+```
+actualFailures:<numberoffailures>
+totalTests:<totalnumberoftests>
+expectedFailures:<expectednumberoffailures>
+```
+
+The build will fail to proceed to the install step if `actualFailures` is greater than `expectedFailures`.
+
+Here is an example implementation of `zopen_check_results()`:
+
+```bash
+zopen_check_results()
+{
+chk="$1/$2_check.log"
+
+failures=$(grep ".* Test.*in .* Categories Failed" ${chk} | cut -f1 -d' ')
+totalTests=$(grep ".* Test.*in .* Categories Failed" ${chk} | cut -f5 -d' ')
+
+cat <<ZZ
+actualFailures:$failures
+totalTests:$totalTests
+expectedFailures:0
+ZZ
+}
+```
+
+`zopen build` will generate a .env file in the install location with support for environment variables such as PATH, LIBPATH, and MANPATH.
+To add your own, you can append environment variables by echo'ing them in a function called `zopen_append_to_env()`.
+
+After the build is successful, `zopen build` will install the project to `$HOME/zopen/prod/projectname`. To perform post-processing on the installed contents, such as modifying hardcoded path contents, you can write a `zopen_post_install()` function which takes the installed path as the first argument.
+
+Note that you can choose the fully-qualified environment variables ZOPEN_GIT_URL, ZOPEN_GIT_DEPS and ZOPEN_TARBALL_URL, ZOPEN_TARBALL_DEPS 
+accordingly if you prefer. See (https://github.com/ZOSOpenTools/zotsampleport/blob/main/setenv.sh) for an example.
+
+There are several additional environment variables that can be specified to provide finer-grained control of the build process. 
+For details
+- Run `zopen build -h` for a description of all the environment variables
+- Read the code: (https://github.com/ZOSOpenTools/meta/blob/main/bin/zopen-build). 
+
+For a sample port, visit the [zotsampleport](https://github.com/ZOSOpenTools/zotsampleport) repo.
+
+#### Running zopen build
+
+Run `zopen build` from the root directory of the git repo you would like to build.  For example, m4:
+```
+cd ${HOME}/zopen/dev/m4
+zopen build
+```
+
+### zopen generate
+You can generate a zopen template project with `zopen generate`. It will ask you a series of questions and then generate the zopen file structure, including a `buildenv` file that will help you get started with your project.
+
+### zopen-importenvs
+This script is used to source the .env from zopen prod/boot directory as explained below:
+
+Usage of script is: ". ./zopen-importenvs [path to buildenv to fetch dependency]"
```

Comment:
can you describe why someone would use this? Is it so they can set up their interactive shell in the same style as what the build would do?

Reply:
Added some more content here